### ISSUE TITLE
Oak-D DepthAI with OpenCV 4.9.0 on Windows

### ISSUE THREAD

**Issue by cesarpgouveia** [opencv/opencv#25164]
### System Information

OpenCV version: 4.9.0
Operating System: Windows 10
depthai-core: https://github.com/luxonis/depthai-core/releases/tag/v2.24.0

### Detailed description

I'm attempting to utilize OpenCV with Oak-D integration (depthai-core), and to do so, I'm following this tutorial: https://docs.opencv.org/4.9.0/df/d88/tutorial_gapi_oak_devices.html

However, upon running CMake with the -DWITH_OAK=ON flag, I don't see any mention of "oak" in the CMake log. It seems like it's not even attempting to search for it. Could you provide some assistance?

### Steps to reproduce

This is the cmake flags that I'm using:

cmake .. -DBUILD_opencv_apps=OFF -DBUILD_opencv_aruco=ON -DBUILD_opencv_bgsegm=OFF -DBUILD_opencv_bioinspired=OFF -DBUILD_opencv_calib3d=ON -DBUILD_opencv_ccalib=ON -DBUILD_opencv_core=ON -DBUILD_opencv_datasets=OFF -DBUILD_opencv_dnn=ON -DBUILD_opencv_dnn_objdetect=OFF -DBUILD_opencv_dnn_superres=OFF -DBUILD_opencv_dpm=OFF -DBUILD_opencv_face=OFF -DBUILD_opencv_features2d=ON -DBUILD_opencv_flann=ON -DBUILD_opencv_fuzzy=OFF -DBUILD_opencv_gapi=OFF -DBUILD_opencv_hfs=OFF -DBUILD_opencv_highgui=ON -DBUILD_opencv_img_hash=ON -DBUILD_opencv_imgcodecs=ON -DBUILD_opencv_imgproc=ON -DBUILD_opencv_intensity_transform=OFF -DBUILD_java_bindings_generator=OFF -DBUILD_opencv_js=OFF -DBUILD_opencv_line_descriptor=OFF -DBUILD_opencv_mcc=OFF -DBUILD_opencv_ml=ON -DBUILD_opencv_objc_bindings_generator=OFF -DBUILD_opencv_objdetect=ON -DBUILD_opencv_optflow=OFF -DBUILD_opencv_phase_unwrapping=OFF -DBUILD_opencv_photo=ON -DBUILD_opencv_plot=ON -DBUILD_opencv_python3=OFF -DBUILD_opencv_python_bindings_generator=OFF -DBUILD_opencv_python_tests=OFF -DBUILD_opencv_quality=OFF -DBUILD_opencv_rapid=OFF -DBUILD_opencv_reg=OFF -DBUILD_opencv_rgdb=OFF -DBUILD_opencv_saliency=OFF -DBUILD_opencv_shape=OFF -DBUILD_opencv_stereo=OFF -DBUILD_opencv_stitching=OFF -DBUILD_opencv_structured_light=OFF -DBUILD_opencv_superres=OFF -DBUILD_opencv_surface_matching=OFF -DBUILD_opencv_text=OFF -DBUILD_opencv_tracking=ON -DBUILD_opencv_ts=OFF -DBUILD_opencv_video=ON -DBUILD_opencv_videoio=ON -DBUILD_opencv_videostab=OFF -DBUILD_opencv_world=ON -DBUILD_opencv_xfeatures2d=ON -DBUILD_opencv_ximgproc=ON -DBUILD_opencv_xobjdetect=ON -DBUILD_opencv_xphoto=ON -DCMAKE_CONFIGURATION_TYPES=Release -DCMAKE_GENERATOR_PLATFORM=x64 -T v142 -DWITH_FFMPEG=ON -DWITH_INF_ENGINE=ON -DENABLE_CXX11=ON -DOPENCV_ENABLE_NONFREE=OFF -DBUILD_JAVA=OFF -DWITH_MSMF=OFF -DWITH_MSMF_DXVA=OFF -DOPENCV_EXTRA_MODULES_PATH="C:/Users/cesar.gouveia/Projects/OpenCV-Package/opencv_contrib/modules" -DBUILD_ZLIB=ON -DBUILD_opencv_hdf=OFF **-DWITH_OAK=ON**

### Issue submission checklist

- [X] I report the issue, it's not a question
- [X] I checked the problem with documentation, FAQ, open issues, forum.opencv.org, Stack Overflow, etc and have not found any solution
- [X] I updated to the latest OpenCV version and the issue is still there
- [X] There is reproducer code and related data files (videos, images, onnx, etc)

**Comment by cesarpgouveia** [comment]
Yes, however, on your tutorial page, you mention "see a status WITH_OAK in CMake log," yet I can't find anything related to this in the CMake log. This is why I suspect I may not have correctly caught or configured Oak-D (DepthAi-core).

**Comment by cesarpgouveia** [comment]
So based on your suggestion I changed my cmake to something like this:

`cmake .. -DBUILD_opencv_apps=OFF -DBUILD_opencv_aruco=ON -DBUILD_opencv_bgsegm=OFF -DBUILD_opencv_bioinspired=OFF -DBUILD_opencv_calib3d=ON -DBUILD_opencv_ccalib=ON -DBUILD_opencv_core=ON -DBUILD_opencv_datasets=OFF -DBUILD_opencv_dnn=ON -DBUILD_opencv_dnn_objdetect=OFF -DBUILD_opencv_dnn_superres=OFF -DBUILD_opencv_dpm=OFF -DBUILD_opencv_face=OFF -DBUILD_opencv_features2d=ON -DBUILD_opencv_flann=ON -DBUILD_opencv_fuzzy=OFF **-DBUILD_opencv_gapi=ON** -DBUILD_opencv_hfs=OFF -DBUILD_opencv_highgui=ON -DBUILD_opencv_img_hash=ON -DBUILD_opencv_imgcodecs=ON -DBUILD_opencv_imgproc=ON -DBUILD_opencv_intensity_transform=OFF -DBUILD_java_bindings_generator=OFF -DBUILD_opencv_js=OFF -DBUILD_opencv_line_descriptor=OFF -DBUILD_opencv_mcc=OFF -DBUILD_opencv_ml=ON -DBUILD_opencv_objc_bindings_generator=OFF -DBUILD_opencv_objdetect=ON -DBUILD_opencv_optflow=OFF -DBUILD_opencv_phase_unwrapping=OFF -DBUILD_opencv_photo=ON -DBUILD_opencv_plot=ON -DBUILD_opencv_python3=OFF -DBUILD_opencv_python_bindings_generator=OFF -DBUILD_opencv_python_tests=OFF -DBUILD_opencv_quality=OFF -DBUILD_opencv_rapid=OFF -DBUILD_opencv_reg=OFF -DBUILD_opencv_rgdb=OFF -DBUILD_opencv_saliency=OFF -DBUILD_opencv_shape=OFF -DBUILD_opencv_stereo=OFF -DBUILD_opencv_stitching=OFF -DBUILD_opencv_structured_light=OFF -DBUILD_opencv_superres=OFF -DBUILD_opencv_surface_matching=OFF -DBUILD_opencv_text=OFF -DBUILD_opencv_tracking=ON -DBUILD_opencv_ts=OFF -DBUILD_opencv_video=ON -DBUILD_opencv_videoio=ON -DBUILD_opencv_videostab=OFF -DBUILD_opencv_world=ON -DBUILD_opencv_xfeatures2d=ON -DBUILD_opencv_ximgproc=ON -DBUILD_opencv_xobjdetect=ON -DBUILD_opencv_xphoto=ON -DCMAKE_CONFIGURATION_TYPES=Release -DCMAKE_GENERATOR_PLATFORM=x64 -T v142 -DWITH_FFMPEG=ON -DWITH_INF_ENGINE=ON -DENABLE_CXX11=ON -DOPENCV_ENABLE_NONFREE=OFF -DBUILD_JAVA=OFF -DWITH_MSMF=OFF -DWITH_MSMF_DXVA=OFF -DOPENCV_EXTRA_MODULES_PATH="C:/Users/cesar.gouveia/Projects/OpenCV-Package/opencv_contrib/modules" -DBUILD_ZLIB=ON -DBUILD_opencv_hdf=OFF **-DWITH_OAK=ON**`

However, on your tutorial page, you mention:

>  "see a status WITH_OAK in CMake log,"

 yet I can't find anything related to this in the CMake log. This is why I suspect I may not have correctly caught or configured Oak-D (DepthAi-core).

**Comment by cesarpgouveia** [comment]
So update it seems to be compiled with GAPI Oak-D support, because I managed to build a project and call your example functions. Yet there's no indication of this in the CMake log of that (but let's forget that). Consequently, I proceeded to build it and essentially replicated your code to evaluate several models. I conducted tests with both ONNX (.onnx) and OpenVINO (.xml, .bin) model formats, but consistently encountered the following error:

```
OpenCV: terminate handler is called! The last OpenCV error is:
OpenCV(4.9.1) Error: Unspecified error (Built without OAK support) in cv::gapi::oak::backend, file C:\Users\cesar.gouveia\Projects\OpenCV-Package\opencv_mirror\modules\gapi\src\backends\oak\goakbackend.cpp, line 1125
```

**Comment by asmorkalov** [comment]
@TolyaTalamanov could you help here?
